edge-modules/TwinTester/TwinOperator.cs

```diff
@@ -0,0 +1,349 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
Since there are two distinct flows here, we can think about splitting it up into different classes. This one is a bit large.

Reply:
I made an interface for twin operations that has perform() and validate(). Good suggestion!